src/services/RPCSelector.ts

```diff
@@ -42,17 +41,26 @@ class RPCSelector {
     return this.getProviderWithHighestBlockNumber(providerComparands);
   }
 
-  private async isProviderUpToDate(url: string): Promise<boolean> {
-    try {
-      const provider = providers.getDefaultProvider(url);
-      const block = <{ timestamp: number }>await Promise.race([provider.getBlock('latest'), this.timeout()]);
-      return isTimestampMoreRecentThan(block.timestamp, this.config.maxTimestampDiff / 1000);
-    } catch {
-      return false;
-    }
+  private getProvidersTimestamps(): Promise<ComparandWithTimestamp>[] {
+    return this.urls.map(async (url) => {
+      try {
+        const provider = providers.getDefaultProvider(url);
+        const block = <{ timestamp: number }>await Promise.race([provider.getBlock('latest'), this.timeout()]);
```

Comment:
```suggestion
        const block: { timestamp: number } = await Promise.race([provider.getBlock('latest'), this.timeout()]);
```
a bit prettier imo

Reply:
Won't work because `timeout` doesn't return anything.
It has to be a type cast.
I can use `as { timestamp: number }` as a workaround.